player/command.c

```diff
@@ -7744,6 +7744,19 @@ void mp_option_change_callback(void *ctx, struct m_config_option *co, int flags,
     if (flags & UPDATE_DEMUXER)
         mpctx->demuxer_changed = true;
 
+    if (flags & UPDATE_AD) {
+        uninit_audio_chain(mpctx);
+        reinit_audio_chain(mpctx);
+    }
+
+    if (flags & UPDATE_VD) {
+        struct track *track = mpctx->current_track[0][STREAM_VIDEO];
+        uninit_video_chain(mpctx);
+        reinit_video_chain(mpctx);
```

Comment:
Do we need full reinint? `UPDATE_HWDEC` does use `VDCTRL_REINIT` to reinit decoder.

Reply:
It doesn't work. The filters need to be recreated.